lib/routes/veeam/list.js

```diff
@@ -32,7 +32,7 @@ function buildXMLResponse(request, arrayOfFiles, versioned = false) {
         Contents: [],
         CommonPrefixes: [],
     };
-    arrayOfFiles.forEach(file => list[versioned ? 'Versions' : 'Contents'].push({
+    const entries = arrayOfFiles.map(file => ({
```

Comment:
Just to keep track of your comments @francoisferrand :

> may be written more expressively like this:

```
const entries = arrayOfFiles.map(file => { ..... })
// Add the folder as the base file
entries.push({ ... }
```

> and that 'same' list of entries can be used to build the appropriate `list` object afterwards, without the error-prone `list[versioned ? 'Versions' : 'Contents']`


-> It's handled here https://github.com/scality/cloudserver/pull/4942/commits/93776d6f12455799f93951ef3da1753303db1439#diff-25506d7604624dfc2c3d66729bd5dffc4534b63a3bffd2d7213ffa1dcb794ae5R35
ALso tested with Artesca